src/api/middleware/offer.js

```diff
@@ -19,7 +19,8 @@ export const isOwnerNotBlocked = async (req, res, next) => {
 };
 
 export const setTargetOwner = (req, res, next) => {
+    const wantedTargetOwner = req.hasAdminPrivileges && req.body.owner;
+    req.targetOwner = req.user?.company?._id.toString() || wantedTargetOwner || undefined;
```

Comment:
Why is the `|| undefined` necessary? Is this different from what we had before?

Reply:
Yes. Please note that before the condition was `req.user?.company?._id.toString() || req.body.owner`. Therefore, since the company ID and the req.body.owner can only be either strings or undefined, the resulting value is a string or undefined.

With the new fix, if the company ID is undefined and the user doesn't have admin privileges, the resulting value would be `false` instead of `undefined`.

The `|| undefined` is there to make sure that the behavior, in the case of non-authorization, stays exactly the same.